### PR TITLE
Jooa11y plugin correction

### PR DESCRIPTION
**Pull request by Denitz**
### Summary of Changes

1. Disable page cache on Jooa11y active state
2. Disable system caching on Jooa11y active state.
3. Load plugin language only if plugin actually runs.
4. Remove useless `$direction` and `$lang` vars.
5. Prettify `isAuthorisedDisplayChecker()` method

### Testing Instructions

Enable Joomla system cache plugin.
Ensure that debug is disabled.
Clear Joomla cache.
Load frontend article as guest.
See cache file in /administrator/cache/page folder
Edit article in backend and perform "Accessibility Check".
See extra file in /administrator/cache/page folder

### Actual result BEFORE applying this Pull Request

An extra cache file is generated for each article

### Expected result AFTER applying this Pull Request

Cache file is not generated.

### Documentation Changes Required

No.

### CODE REVIEW

File: plugins/system/jooa11y/jooa11y.php [hunk 1]
```diff
@@ -105,23 +91,11 @@ private function isAuthorisedDisplayChecker(): bool
...
-		if (!$showJooa11y || !$this->isAuthorisedDisplayChecker())
-		{
-			return;
-		}
-
 		// Get the document object.
 		$document = $this->app->getDocument();
 
-		// Determine if it is an LTR or RTL language
-		$direction = Factory::getLanguage()->isRtl() ? 'right' : 'left';
```
Comment (brianteeman) [inline plugins/system/jooa11y/jooa11y.php]: Good spot. When I updated the css to use logical properties this should have been removed

File: plugins/system/jooa11y/jooa11y.php [hunk 2]
```diff
@@ -260,14 +234,46 @@ public function initJooa11y()
...
 		$wa = $document->getWebAssetManager();
 
 		$wa->getRegistry()->addRegistryFile('media/plg_system_jooa11y/joomla.asset.json');
 
 		$wa->useScript('plg_system_jooa11y.jooa11y')
 			->useStyle('plg_system_jooa11y.jooa11y');
+	}
+
+	/**
+	 * Method to check if the current user is allowed to see the debug information or not.
```
Comment (brianteeman) [inline plugins/system/jooa11y/jooa11y.php]:
```suggestion
	 * Method to check if the current user is allowed to see the checker or not.
```

I know this was an existing bug but might as well get fixed here at the same time

File: plugins/system/jooa11y/jooa11y.php [hunk 3]
```diff
@@ -1,10 +1,10 @@
 <?php
 /**
- * @package     Joomla.Plugin
- * @subpackage  System.jooa11y
+ * @package         Joomla.Plugin
+ * @subpackage      System.jooa11y
```
Comment (brianteeman) [inline plugins/system/jooa11y/jooa11y.php]: please revert this chanage

File: plugins/system/jooa11y/jooa11y.php [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
  *
  * @copyright   (C) 2021 Open Source Matters, Inc. <https://www.joomla.org>
- * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ * @license         GNU General Public License version 2 or later; see LICENSE.txt
```
Comment (brianteeman) [inline plugins/system/jooa11y/jooa11y.php]: please revert this change

Comment (Denitz) [reply plugins/system/jooa11y/jooa11y.php]: Done!

File: plugins/system/jooa11y/jooa11y.php (hunk 3 above)
Comment (Denitz) [reply plugins/system/jooa11y/jooa11y.php]: Done!

File: plugins/system/jooa11y/jooa11y.php [hunk 5]
```diff
@@ -30,22 +30,14 @@ class PlgSystemJooa11y extends CMSPlugin implements SubscriberInterface
...
-	 * @since  4.1.0
-	 */
-	protected $autoloadLanguage = true;
-
 	/**
 	 * Subscribe to certain events
 	 *
 	 * @return string[]  An array of event mappings
 	 *
+	 * @throws Exception
```
Comment (Quy) [inline plugins/system/jooa11y/jooa11y.php]: Please revert. See https://developer.joomla.org/coding-standards/docblocks.html

Comment (Denitz) [reply plugins/system/jooa11y/jooa11y.php]: @Quy Done!

**Comment by brianteeman** [comment]
@Denitz you do not need to keep updating your branch. Unless there are conflicts to fix its best if you dont as every commit will reset the test count

**Comment by Denitz** [comment]
Argh, sorry, I didn't know about test count.

**Comment by carcam** [comment]
I have tested this item :red_circle: unsuccessfully on 74b13e59085656fb54e014f79d8d4fd5861ef46d<br /><br />I'm afraid I cannot reproduce the issue.

I have tried this:

1. Enabling conservative caching
1. Enabling both conservative caching and page cache

Accesibility button only appears in frontend when the "show always" check is marked on the plugin configuration.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37068">issues.joomla.org/tracker/joomla-cms/37068</a>.</sub>

**Comment by Denitz** [comment]
@carcam Do you have Joomla debug enabled? Caching is not working in this case, should be disabled.

**Comment by carcam** [comment]
Hi @Denitz thank you for your reply.

Debug is disabled and attached you may see some relevant screenshots of my current configuration.

Captura de Pantalla 2022-05-27 a las 16 44 56
Captura de Pantalla 2022-05-27 a las 16 44 46

Captura de Pantalla 2022-05-27 a las 16 45 49
Captura de Pantalla 2022-05-27 a las 16 45 35
Captura de Pantalla 2022-05-27 a las 16 45 25
Captura de Pantalla 2022-05-27 a las 16 45 16

Still no luck reproducing the issue in latest code from 4.1-dev branch...

**Comment by Denitz** [comment]
@carcam Not sure what's wrong, but it's really not replicated.

Moreover, assuming that Jooa11y injects assets via `onBeforeCompileHead` event, these assets can't be cached via view cache (caching in global config) for guests.

But, it's still cached with 'System - Page Cache' enabled for accessibility check preview URL like https://site.com/news/article-alias?jooa11y=1 (see ?jooa11y=1 appended). Having an extra cache file is useless.

I've updated summary of changes and test instructions.

**Comment by Denitz** [comment]
@carcam I found why it's not reproduceable anymore: recently, we have pushed the modules cache update and now the module cache only contains assets added by these module, but not all assets existing at the time of module generation.
Before, the article was rendered and cached, next template is rendered, Jooa11y adds own assets via onBeforeCompileHead, next modules were rendered and cached (and the module cache contained Jooa11y assets).
Anyway, the cache should be disabled on Jooa11y active state.

**Comment by carcam** [comment]
I have tested this item :white_check_mark: successfully on 74532452abe6de599be00cfc11108f00f294046b<br /><br />Hi @Denitz 
Thank you very much for rewriting the instructions!!
I have been able to reproduce the issue and after applying the patch no files appear in the cache folder in any case. If that was the intention everything is fine ;).<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37068">issues.joomla.org/tracker/joomla-cms/37068</a>.</sub>

**Comment by Quiviro** [comment]
I have not tested this item.<br /><br />It seems I cannot reproduce the issue.

If I open the article as guest, it creates cache file. Then, when I edit and check in backoffice the extra cache file is not generated (after and before applying patch).
I copied the configuration from carcam<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37068">issues.joomla.org/tracker/joomla-cms/37068</a>.</sub>

**Comment by laoneo** [comment]
@brianteeman can you have a look here again if all is ok?

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 74532452abe6de599be00cfc11108f00f294046b<br /><br />Cannot replicate the reported problem



Load frontend article as guest.
See cache file in /administrator/cache/page folder - **_YES_**
Edit article in backend and perform "Accessibility Check".
See extra file in /administrator/cache/page folder - **_NO_**<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/37068">issues.joomla.org/tracker/joomla-cms/37068</a>.</sub>

**Comment by Denitz** [comment]
@brianteeman Please ensure that system page cache plugin is enabled and you are not logged in frontend.

**Comment by brianteeman** [comment]
pretty sure i set it up exactly as shown in @carcam screenshots but will double check

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Comment by obuisard** [comment]
Could you make 2 pull requests? One for the refactoring and one for the cache issue? Would be great, thanks.

**Comment by Denitz** [comment]
Sorry, I can't, this PR is simple enough.

**Comment by HLeithner** [comment]
This pull request has been automatically rebased to 4.4-dev.

**Comment by Denitz** [comment]
Closed in favour of #41956
